### DISCUSSION TITLE
Developer changes in existing strings not being picked up by weblate. Is this intended behavior?

### DISCUSSION THREAD

**Discussion by chrisandrewcl** [WeblateOrg/weblate#10092] [Q&A]
I am using Weblate for a new project and noticed that an existing string I've changed in en-US.json (our "source" locale) is outdated in Weblate. New ones were correctly added and marked as "needs editing", but the existing ones are still showing the older values and there is no indication that they have been updated. My expectation was that it would receive the new values and be marked as "needs editing".

\* This project uses monolingual JSON files.

Is this intended behavior? If so, should we never change strings directly on the files and instead edit only in Weblate? 

(I am hoping this is not the case)

Do you have any recommendations for this workflow?

**Comment by nijel** [comment]
String updated in the repository should be updated when Weblate parses updated files.

**Reply by chrisandrewcl** [reply]
When does the parsing happen? Is it at the same time when new strings are added or is it a background job that maybe hasn't fired yet, and if so, is there any way to manually trigger it or any logs to look into why the update didn't happen? 

I waited from 20 minutes to one hour after Weblated received the commit before I went and manually updated them in Weblate to keep the process going. 

\* I see that I am running an outdated version of Weblate (4.8.1). Is this maybe a known issue that was fixed in a newer version?

**Reply by nijel** [reply]
Incoming webhooks are handled in the background, so there might be some delay. You should see progress in the server logs in this case.

Using outdated version should not be an issue in this, this behavior hasn't changed for years. There might be some related issues which got fixed meanwhile, though...

**Comment by emmapeel2** [comment]
Did you enable the `Cleanup translation files` add-on? I think this is needed for the json translations to be updated.

**Reply by chrisandrewcl** [reply]
Yes, the said add-on is enabled. Here is the complete list: 

-  Remove blank strings 
-  Flag unchanged translations as "Needs editing" 
-  Flag new source strings as "Needs editing" 
-  Flag new translations as "Needs editing" 
-  Cleanup translation files 
-  Squash Git commits 
-  Customize JSON output

**Comment by chrisandrewcl** [comment]
It seems there is something definitely odd here. This is the history being shown now in Weblate for a specific string affected by the issue:

![odd-history](https://github.com/WeblateOrg/weblate/assets/368831/af558b53-b2e2-4505-9d98-e93718047d23)

1. The diff for the first change (3), made over one month ago on weblate, doesn't match with the resulting string for that change, and the resulting string is not what was on Weblate at that time, but instead the final string which would be pulled from the repository via webhook in the future.
3. The second change (2) was me manually changing the string to match what was in the repository and marking "Needs editing" a few minutes after the webhook fired and I noticed it failed to update it.
5. The last change (3) was my colleague just unchecking "Needs editing".
